osu.Game/Overlays/BeatmapSet/BeatmapSetHeaderContent.cs

```diff
@@ -268,11 +268,13 @@ private void updateDownloadButtons()
                     break;
 
                 case DownloadState.Downloading:
-                case DownloadState.Importing:
                     // temporary to avoid showing two buttons for maps with novideo. will be fixed in new beatmap overlay design.
                     downloadButtonsContainer.Child = new HeaderDownloadButton(BeatmapSet.Value);
                     break;
 
+                case DownloadState.Importing:
+                    break;
```

Comment:
I'm not especially convinced by this change. I understand the reasoning behind it, but as far as I can tell it relies on an assumption that an `Importing` state will only ever follow a `Downloading` state, which I'm not sure if it's a valid assumption. May be safer to keep the previous code but also check the count of buttons in `downloadButtonsContainer`.

Reply:
Right, sounds like it would be the case when importing a beatmap without downloading it from the overlay.

Reply:
I checked that case and it isn't a problem since `IModelDownloader` only has callbacks on download began/failed and item updated/removed (via `IModelManager`). I think there is another theoretical path that could trigger this, which is:

* start download from beatmap listing
* move to beatmap overlay by clicking on the beatmap card

[There is an explicit code path that skips `Downloading`](https://github.com/ppy/osu/blob/a2215d8078efd9f216b9392ade7df60982626796/osu.Game/Online/DownloadTrackingComposite.cs#L124-L128), but I haven't managed to get it to actually trigger.

Reply:
> May be safer to keep the previous code but also check the count of buttons in downloadButtonsContainer.

The animations will be visually different for non-video beatmaps, as the button won't be recreated.

![Kapture 2021-04-27 at 18 36 16](https://user-images.githubusercontent.com/35318437/116332919-76c55700-a787-11eb-9a4e-b2493e452faa.gif)
![Kapture 2021-04-27 at 18 36 31](https://user-images.githubusercontent.com/35318437/116332936-7d53ce80-a787-11eb-8f27-f3f196f35b0c.gif)

I think it's best just to revert this change, and make these animations work without recreating anything.